crates/svm-runtime/tests/wasm/runtime_func_buf.wast

```diff
@@ -1,17 +1,80 @@
 (module
-  (func $get64 (import "svm" "get64") (param i32) (result i64))
-  (func $set64 (import "svm" "set64") (param i32 i64))
+  (func $store160 (import "svm" "store160") (param $mem_idx i32) (param $mem_ptr i32) (param $var_id i32))
+  (func $load160 (import "svm" "load160") (param $var_id i32) (param $mem_idx i32) (param $mem_ptr i32))
 
   (import "svm" "memory" (memory 1))
 
   (func (export "ctor")
   	nop)
 
-  (func (export "copy64") (param $var_id i32) (param $mem_ptr i32)
-  	get_local $var_id
+  (func (export "store_addr") (param $var_id i32) (param $mem_ptr i32)
+  	i32.const 0         ;; mem_idx
+	get_local $mem_ptr  ;; mem_ptr
+  	get_local $var_id   ;; var_id
+	call $store160)
 
-  	get_local $mem_ptr
-	i64.load
+  (func (export "edit_addr") (param $func_buf_size i32) (param $var_id i32) 
+	(local $i i32) 
 
-	call $set64))
-  	
\ No newline at end of file
+	;; we load var `var_id` from storage starting from memory address `func_buf_size`.
+	;; since the `func_buf` consumes memory range: `[0...func_buf_size)`
+	;; we know that loading var `var_id` won't collide with `func_buf`.
+
+  	get_local $var_id        ;; var_id
+  	i32.const 0              ;; mem_idx
+  	get_local $func_buf_size ;; mem_ptr
+	call $load160
+
+        ;; now we'll start editing var `var_id`
+
+	;; i <- 0
+        (set_local $i (i32.const 0))
+
+	(block 
+	    (loop 
+		;; func_buf_size + i
+		get_local $func_buf_size
+		get_local $i
+		i32.add
+
+		;; byte <- func_buf[i] 
+		;; (i.e reading from `mem[i]` since `func_buf` starts at `address=0`)
```

Comment:
Maybe "starts at `index=0`"? using the term `address` here is a bit confusing.

Reply:
Using `index=0` seems to me a bit more obscure. Because it's actually really is a memory `address = 0`.

Reply:
I guess I got confuse because the example was dealing with a byte array of an address. 

I just looked at WebAssembly specifications, and there it's called `offset`. Maybe we should use that.